planning/behavior_velocity_planner/src/scene_module/stop_line/scene.cpp

```diff
@@ -81,46 +81,88 @@ bool StopLineModule::modifyPathVelocity(
-    planning_utils::insertStopPoint(stop_pose.position, stop_line_seg_idx, *path);
-
-    // Update first stop index
-    first_stop_path_point_index_ = static_cast<int>(stop_point_idx);
-    debug_data_.stop_pose = stop_pose;
-
-    // Get stop point and stop factor
-    {
-      tier4_planning_msgs::msg::StopFactor stop_factor;
-      stop_factor.stop_pose = stop_pose;
-      stop_factor.stop_factor_points.push_back(getCenterOfStopLine(stop_line_));
-      planning_utils::appendStopReason(stop_factor, stop_reason);
-    }
+  switch (state_) {
+    case State::APPROACH: {
+      // Insert stop pose
+      planning_utils::insertStopPoint(stop_pose.position, stop_line_seg_idx, *path);
+
+      // Update first stop index
+      first_stop_path_point_index_ = static_cast<int>(stop_point_idx);
+      debug_data_.stop_pose = stop_pose;
+
+      // Get stop point and stop factor
+      {
+        tier4_planning_msgs::msg::StopFactor stop_factor;
+        stop_factor.stop_pose = stop_pose;
+        stop_factor.stop_factor_points.push_back(getCenterOfStopLine(stop_line_));
+        planning_utils::appendStopReason(stop_factor, stop_reason);
+      }
+
+      // Move to stopped state if stopped
+      if (
+        signed_arc_dist_to_stop_point < planner_param_.hold_stop_margin_distance &&
+        planner_data_->isVehicleStopped()) {
+        RCLCPP_INFO(logger_, "APPROACH -> STOPPED");
+
+        state_ = State::STOPPED;
+        stopped_time_ = std::make_shared<const rclcpp::Time>(clock_->now());
 
-    // Move to stopped state if stopped
-    if (
-      signed_arc_dist_to_stop_point < planner_param_.stop_check_dist &&
-      planner_data_->isVehicleStopped(planner_param_.stop_duration_sec)) {
-      RCLCPP_INFO(logger_, "APPROACH -> STOPPED");
-      state_ = State::STOPPED;
-      if (signed_arc_dist_to_stop_point < -planner_param_.stop_check_dist) {
-        RCLCPP_ERROR(
-          logger_, "Failed to stop near stop line but ego stopped. Change state to STOPPED");
+        if (signed_arc_dist_to_stop_point < -planner_param_.hold_stop_margin_distance) {
+          RCLCPP_ERROR(
+            logger_, "Failed to stop near stop line but ego stopped. Change state to STOPPED");
+        }
       }
+
+      break;
     }
-  } else if (state_ == State::STOPPED) {
-    // Change state after vehicle departure
-    if (!planner_data_->isVehicleStopped()) {
-      RCLCPP_INFO(logger_, "STOPPED -> START");
-      state_ = State::START;
+
+    case State::STOPPED: {
+      // Change state after vehicle departure
+      const auto stopped_pose = motion_utils::calcLongitudinalOffsetPose(
+        path->points, planner_data_->current_pose.pose.position, 0.0);
+
+      if (!stopped_pose) {
+        break;
+      }
+
+      SegmentIndexWithPose ego_pos_on_path;
+      ego_pos_on_path.pose = stopped_pose.get();
+      ego_pos_on_path.index = motion_utils::findFirstNearestSegmentIndexWithSoftConstraints(
+        path->points, stopped_pose.get());
```

Comment:
use findEgoSegmentIndex which is a wrapper function of findFirstNearestSegmentIndexWithSoftConstraints with proper praramters.

Reply:
Fix in https://github.com/autowarefoundation/autoware.universe/pull/1648/commits/0527bba1710ce7d7a8bbd6c1d55da6268b42357c